persistence/module.go

```diff
@@ -102,7 +104,7 @@ func (*persistenceModule) Create(bus modules.Bus, options ...modules.ModuleOptio
 		return nil, err
 	}
 
-	stateTrees, err := newStateTrees(persistenceCfg.TreesStoreDir)
+	stateTrees, err := trees.NewStateTrees(persistenceCfg.TreesStoreDir, txIndexer)
```

Comment:
Thoughts on having the `TreeStore` be a submodule (using dependency injection, being modular, etc...) that takes in a bus? Then you can access the `txIndexer` via `GetBus().GetPersistenceModule().GetTxIndexer()`

Reply:
Totally open to it, is it okay if i address it in a follow up ticket?

Reply:
Addressed in #808